core/src/main/scala/org/apache/spark/storage/DiskBlockObjectWriter.scala

```diff
@@ -27,8 +27,8 @@ import org.apache.spark.util.Utils
 
 /**
  * A class for writing JVM objects directly to a file on disk. This class allows data to be appended
- * to an existing block and can guarantee atomicity in the case of faults as it allows the caller to
- * revert partial writes.
+ * to an existing block. Callers can write to the same file and commit these writes.
+ * In case of faults, callers should atomically revert the uncommitted partial writes.
```

Comment:
Perhaps elaborate a bit more, e.g. "For efficiency, this class retains the underlying file channel across multiple commits to a file. The channel is kept open until close() is called on DiskBlockObjectWriter."